src/storage/src/render/mod.rs

```diff
@@ -7,98 +7,198 @@
-//! oks stream could represent the correct result of the computation where all
-//! rows that caused an error have been pruned from the stream. There are
-//! strange and confusing questions here around foreign keys, though: what if
-//! the optimizer proves that a particular key must exist in a collection, but
-//! the key gets pruned away because its row participated in a scalar expression
-//! evaluation that errored?
-//!
-//! In the meantime, it is probably wise for operators to keep the oks stream
-//! roughly "as correct as possible" even when errors are present in the errs
-//! stream. This reduces the amount of recomputation that must be performed
-//! if/when the errors are retracted.
+//! #### Reclocking
+//!
+//! Whenever a dataflow edge crosses the scope boundaries it must first be converted into a
+//! captured stream via the `[mz_timely_util::capture::UnboundedTokioCapture`] utility. This
+//! disassociated the stream and its progress information from the original timely scope and allows
+//! it to be read from a different place. The downside of this mechanism is that it's invisible to
+//! timely's progress tracking, but that seems like a necessary evil if we want to do reclocking.
+//!
+//! The two main ways these tokio-fied streams are turned back into normal timely streams in the
+//! destination scope are by the `remap operator` and the `reclock operator` which process the
+//! `data output` and `upper output` of the source reader respectively.
+//!
+//! The `remap operator` reads the `upper output`, which is composed only of frontiers, mints new
+//! bindings, and writes them into the remap shard. The final durable timestamp bindings are
+//! emitted as its output for consumption by the `reclock operator`.
+//!
+//! The `reclock operator` reads the `data output`, which contains both data and progress
+//! statements, and uses the bindings it receives from the `remap operator` to reclock each piece
+//! of data and each frontier statement into the target scope's timestamp and emit the reclocked
+//! stream in its output.
+//!
+//! #### Partitioning
+//!
+//! At this point we have a timely stream with correctly timestamped data in the mz time domain
+//! (`mz_repr::Timestamp`) which contains multiplexed messages for each of the potential subsources
+//! of this source. Each message selects the output it belongs to by setting output field in
+//! [`crate::source::types::SourceMessage`]. By convention, the main source output is always output
+//! zero and subsources get the outputs from one onwards.
+//!
+//! However, regardless of whether the output is the main source or a subsource it is treated
+//! identically by the pipeline. Each output is demultiplexed into its own timely stream using
+//! [`timely::dataflow::operators::partition::Partition`] and the rest of the ingestion pipeline is
+//! rendered independently.
+//!
+//! #### Resumption frontier
```

Comment:
I think this section should be more clear about if the way we communicate this back is through a `Feedback` edge, or not

Reply:
Good catch! Will fix